### PR TITLE
refactor: improve Python package development cycle

### PR DESCRIPTION
**Pull request by Guo-Zhang**
1. `.gitignore`: more complete ignored files; ignore `config.yml`
2. `pyproject.toml`: init with `poetry`
3. `tests`: sample unittest with `pytest`
4. `.github/workflows/testing.yml`: running unittest for each PR or push to main

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,8 +194,7 @@ assets/private.yml
 assets/garbage
 **.log
 log/**
-
-XAgentWeb/src/api/backend.ts
+config.*
```
Comment (Umpire2018) [inline .gitignore:197]: Why do we need `config.*` ?

File: .gitignore [hunk 2]
```diff
@@ -1,23 +1,186 @@
...
+.idea/
+
+# VSCode
+.vscode/
+
+# macOS
+.DS_Store
+
+
+### XAgent
```
Comment (Umpire2018) [inline .gitignore:170]: Maybe change here from `### XAgent` to `# XAgent`

File: .github/workflows/testing.yaml
Comment (Umpire2018) [inline .github/workflows/testing.yaml:1]: Todo: Only use `poetry` to manage all dependency.

File: .gitignore
Comment (Umpire2018) [inline .gitignore:1]: I think there are several parts like `PyBuilder, Sphinx documentation` that are unnecessary for the project.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+name = "xagent"
+# Semantic versioning: https://semver.org
+version = "1.0.0"
+description = "An Autonomous LLM Agent for Complex Task Solving"
+authors = ["XAgent Team <xagentteam@gmail.com>"]
+license = "Apache 2.0"
+readme = "README.md"
+
+[tool.poetry.dependencies]
+# `XAgent/config.py` has `match` syntax, which is from 3.10
```
Comment (Umpire2018) [inline pyproject.toml:16]: What is this for?

File: tests/XAgent/test_config.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from XAgent.config import XAgentConfig
+
+
+def test_xagent_config_to_dict_safely():
+    # 创建一个XAgentConfig实例
+    config = XAgentConfig()
+
+    # 设置实例的属性和数据
```
Comment (Umpire2018) [inline tests/XAgent/test_config.py:8]: Maybe use English comments.

**Review by Umpire2018 — COMMENTED** [review]
First of all, thank you for your contribution. 
This PR implements a simple test and begins managing dependencies with poetry, which is a good start for us. 
However, if it were me, I would have split this PR into different features to keep the PRs to minimal functionality. For example, one PR to introduce Poetry for managing all dependencies, one PR to update the .gitignore, and one PR to add automated tests. 
But I think your PR has set a good example for the project in terms of testing, so I feel it can be merged.
PS. Suggest to use [conventional commits](https://www.conventionalcommits.org/) next time.

File: .gitignore (hunk 1 above)
Comment (Guo-Zhang) [reply .gitignore:197]: The API keys are configured in `config.yml`, so it must be ignored. The better specification is to split default_config, user config and secrets, which should be in the task list.

File: .gitignore (hunk 2 above)
Comment (Guo-Zhang) [reply .gitignore:170]: `###` instead of `#` can be parsed by PyCharm as a section of the ignore file.

File: .github/workflows/testing.yaml
Comment (Guo-Zhang) [reply .github/workflows/testing.yaml:1]: Yeah, it should be considered. But I am not familiar with the dependencies thus cannot set a proper version for them.

File: pyproject.toml (hunk 3 above)
Comment (Guo-Zhang) [reply pyproject.toml:16]: The constraint of Python version, since `match` syntax is used.

File: tests/XAgent/test_config.py (hunk 4 above)
Comment (Guo-Zhang) [reply tests/XAgent/test_config.py:8]: Sure.

**Comment by Guo-Zhang** [comment]
1. Too much PR is no need for this small change. The commits have been spilted into several ones.
2. The PR template is not friendly enough for this PR.
3. The conversional commits of GitHub seems a little different with my format, the Angular's,  (https://gist.github.com/brianclements/841ea7bffdb01346392c).  This one is more popular but complex.

File: .gitignore (hunk 1 above)
Comment (Umpire2018) [reply .gitignore:197]: I get it. But we already ignored those file. I thought this wildcard is going to ignore like `config.toml, config.temp` etc?
```yml
assets/config.yml
assets/gpt4_config.yml
config.yml
```

File: .gitignore (hunk 2 above)
Comment (Umpire2018) [reply .gitignore:170]: What about other section in this file? Like `# VSCode` ?

File: pyproject.toml (hunk 3 above)
Comment (Umpire2018) [reply pyproject.toml:16]: I think this is just one of the reasons. But keep it should be ok.

**Review by Umpire2018 — APPROVED** [review]

**Comment by Umpire2018** [comment]
@Guo-Zhang Let's use conventional commit, i think it's more straightforward.

> 1. Too much PR is no need for this small change. The commits have been spilted into several ones.
> 2. The PR template is not friendly enough for this PR.
> 3. The conversional commits of GitHub seems a little different with my format, the Angular's,  (https://gist.github.com/brianclements/841ea7bffdb01346392c).  This one is more popular but complex.

**Comment by luyaxi** [comment]
Maybe all developers should check the testing workflow and add their own workflow.

**Comment by Umpire2018** [comment]
> Maybe all developers should check the testing workflow and add their own tests.

Add @LOGIC-10

**Review by thuqinyj16 — APPROVED** [review]
Great!

**Review by luyaxi — COMMENTED** [review]
Great

**Review by luyaxi — APPROVED** [review]

**Pull request merged**